### PR TITLE
Project health

### PR DESCRIPTION
**Pull request by peter-budo**
• Fixed checkstyle issues and added some suppressions
• Fixed findbugs, in case you want to get reports locally you can simply execute `gradle clean core:check` if you want html reports instead of xml add following to parent build gradle

```
    tasks.withType(FindBugs) {
        reports {
            xml.enabled = false
            html.enabled = true
        }
    }
```

• Lint is tricky. I may later put suppression on drawable reports. However we have "currently" unused strings that will be used in future and some that are not used at all.  I do strongly advise not to put in commits new strings that are irrelevant to commit and only be used later on. Do put there only stuff that is need it by commit. There are also some colours that are unused and I do remember that some of them are used somewhere by drawer???? We will need to filter through them

### CODE REVIEW

**Comment by peter-budo** [comment]
@karllindmark ready for check build http://ci.peterscorner.co.uk/job/BF4-Intel-PR/28/

**Comment by karllindmark** [comment]
@peter-budo 
Dude, did you not try the app after changing stuff around? All the URLs are now broken, as seen below:

```
http://battlelog.battlefield.com/bf4/UTF-8
```

Steps to reproduce:
1. Select any stats item in the menu
2. Note that it doesn't load
3. Debug and print the URL in, for example, SoldierOverviewFragment
4. Notice that it's the one above, and not what we're expecting

For example, this code below is found in the UrlFactory class:

```
String.format("UTF-8", "warsawoverviewpopulate/%d/%d/", soldierId, platformId)
```

This uses "UTF-8" as the "base string", and all the other arguments as things to insert at specific points.

**Comment by peter-budo** [comment]
@karllindmark Bahhhh I though I pushed that yesterday which look like not.
Here is new build with pushed changes http://ci.peterscorner.co.uk/job/BF4-Intel-PR/29/

PS: We will need to look into espresso or other UI testing framework to avoid such none failing build as previous one

**Comment by karllindmark** [comment]
@peter-budo 
Out of curiosity - I'm not sure we need to clutter up the `String.format(...)` calls with the `Locale.getDefault()`, as that is the default behavior when not specifiying a `Locale`. However, I reckon this is adjusted due to Lint complaining about that? 

As seen in the <a href="http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/String.java#String.format%28java.lang.String%2Cjava.lang.Object%5B%5D%29">Java source</a>:

```
Returns a formatted string using the specified format string and arguments.
The locale always used is the one returned by java.util.Locale.getDefault().
Parameters:
format A format string
args Arguments referenced by the format specifiers in the format string. If there are more arguments than format specifiers, the extra arguments are ignored. The number of arguments is variable and may be zero. The maximum number of arguments is limited by the maximum dimension of a Java array as defined by the Java Virtual Machine Specification. The behaviour on a null argument depends on the conversion.
Returns:
A formatted string
Throws:
IllegalFormatException If a format string contains an illegal syntax, a format specifier that is incompatible with the given arguments, insufficient arguments given the format string, or other illegal conditions. For specification of all possible formatting errors, see the Details section of the formatter class specification.
NullPointerException If the format is null
Since:
1.5
See also:
java.util.Formatter

public static String More ...format(String format, Object ... args) {
    return new Formatter().format(format, args).toString();
}
```

In other words, this:

```
The locale always used is the one returned by java.util.Locale.getDefault().
```

Same goes for <a href="http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/String.java#String.toUpperCase%28%29">String.toUpperCase()</a> and <a href="http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/String.java#String.toLowerCase%28%29">String.toLowerCase()</a>.

Just checking to see if it's necessary, as it does seem unnecessary if it's been the default behavior in Java since pre 1.6 at least. Why Lint is complaining, I don't really know, as I deem it being a better alternative to use the users locale. :-)

**Comment by peter-budo** [comment]
I think this is gone be expected as required declaration in Java 1.8 and the latest code analyses tools do suggest to amend code in that regards. So as @karllindmark put it, if it is Google who use it and suggest it we better do it :smiling_imp:

**Comment by karllindmark** [comment]
@peter-budo: Well, I just wanted to know if there was any reason for Lint complaining on the implicit Locale.getDefault(), but not if you specify Locale.getDefault() explicit in the parameters. :dancer:
